hyperspy/_signals/eds.py

```diff
@@ -549,17 +554,17 @@ def get_lines_intensity(self,
                           "You can remove it with" +
                           "s.metadata.Sample.xray_lines.remove('%s')"
                           % (xray))
-
+        if hasattr(integration_windows, '__iter__') is False:
+            integration_windows = self.estimate_integration_windows(
+                windows_width=integration_windows, xray_lines=xray_lines)
         intensities = []
         # test 1D Spectrum (0D problem)
         # signal_to_index = self.axes_manager.navigation_dimension - 2
-        for Xray_line in xray_lines:
+        for Xray_line, window in zip(xray_lines, integration_windows):
             line_energy, line_FWHM = self._get_line_energy(Xray_line,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
-            det = integration_window_factor * line_FWHM / 2.
-            img = self[..., line_energy - det:line_energy + det
-                       ].integrate1D(-1)
+            img = self[..., window[0]:window[1]].integrate1D(-1)
```

Comment:
I guess here you want to slice the signal? as per #427 , maybe then better to use `.isig[]` explicitly? It's your call in the end, though